src/astro/sql/operators/agnostic_stats_check.py

```diff
@@ -192,9 +188,9 @@ def __init__(
         self.max_rows_returned = max_rows_returned
         self.conn_id = main_table.conn_id
         self.checks = checks
-        self.database = main_table.database
+        self.database = getattr(main_table.metadata, "database", None)
 
-        task_id = main_table.table_name + "_" + "stats_check"
+        task_id = main_table.name + "_" + "stats_check"
```

Comment:
Why not use f-string:

```suggestion
        task_id = f"{main_table.name}_stats_check"
```

f-strings are better for readability